### ISSUE TITLE
Training on custom environment using multiprocessing not printing mean reward during

### ISSUE THREAD

**Issue by atharvaaalok** [DLR-RM/stable-baselines3#1775]
### 🐛 Bug

I have a custom gymnasium environment that works well when I train algorithms (say PPO) and it prints a nice log with mean reward and other quantities when training.

But when I do multiprocessing and train the model in parallel, I do not see the mean reward output.
Why is that and how can I fix that?

### Code example

My code for multiprocessing:
```
def make_env(env_id: str, rank: int, seed: int = 0) -> Callable:

    def _init() -> gym.Env:
        s0 = np.array([[1, 0], [0.75, 0.05], [0.625, 0.075], [0.5, 0.1], [0.25, 0.05], [0, 0], [0.25, -0.05], [0.5, -0.1], [0.625, -0.075], [0.75, -0.05], [1, 0]])
        idx_to_change = [1, 2, 3, 4, 6, 7, 8, 9]
        a_scaling = (1 / 1000)
        valid_states_file_path = os.path.dirname(__file__) + '/Dataset/Arrays_as_rows.txt'

        # Get the environment
        training_env = CFD_Env(s0, idx_to_change, a_scaling, valid_states_file_path)
        # Reset the environment
        observation, info = training_env.reset(seed = seed + rank)

        return training_env

    set_random_seed(seed)
    return _init
```

Then I train the model as follows:
```
env_id = 'CFD_Gym_Env'
    num_cpu = 5
    env = SubprocVecEnv([make_env(env_id, i) for i in range(num_cpu)])


    # Train PPO on the environment
    model = PPO('MlpPolicy', env, verbose = 1, tensorboard_log = log_dir)
```

My custom environment:

```import gymnasium as gym
from gymnasium import spaces

import numpy as np

# Below is my own package for doing some aerodynamics calculations
from ..CFD.Aerodynamics import Aerodynamics


NEGATIVE_REWARD = -100.0


class CFD_Env(gym.Env):
    """Custom Environment that follows gym interface."""

    metadata = {"render_modes": ["human", "no_display"], "render_fps": 4}

    def __init__(self, s0, idx_to_change, a_scaling, valid_states_file_path):
        super(CFD_Env, self).__init__()
        self.action_space = spaces.Box(low = -1, high = 1, shape = (len(idx_to_change) * 2,), dtype = np.float32)
        self.observation_space = spaces.Box(low = -1, high = 2, shape = s0.flatten().shape, dtype = np.float32)

        self.s = s0
        self.idx_to_change = idx_to_change
        self.a_scaling = a_scaling
        self.valid_states_file_path = valid_states_file_path
        self.iter = 0


    def step(self, action):
        action = action.reshape(-1, 2)
        s_new = self.s
        s_new[self.idx_to_change, :] = s_new[self.idx_to_change, :] + action * self.a_scaling
        self.s = s_new

        terminated = False
        truncated = False

        # Generate reward
        airfoil_name = 'air' + str(np.random.rand(1))[3:-1]
        airfoil_coordinates = s_new
        airfoil = Aerodynamics.Airfoil(airfoil_coordinates, airfoil_name)
        Reynolds_num = 1e6
        L_by_D_ratio = airfoil.get_L_by_D(Reynolds_num)

        if L_by_D_ratio == None:
            L_by_D_ratio = NEGATIVE_REWARD
            terminated = True
        
        observation = s_new.flatten()
        reward = L_by_D_ratio
        info = {}

        self.iter += 1
        if self.iter == 80:
            terminated = True
        else:
            terminated = False

        return observation, reward, terminated, truncated, info

    def reset(self, seed = None, options = None):
        np.random.seed(seed)
        # Choose a random valid initial state from file
        file_path = self.valid_states_file_path
        s_new = read_random_line(file_path).reshape(-1, 2).astype(np.float32)

        self.s = s_new

        observation = s_new.flatten()
        # observation = s_new[self.idx_to_change, :].flatten()
        info = {}

        self.iter = 0

        return observation, info

    def render(self):
        pass

    def close(self):
        pass




# Function to read a random line from the file and convert it into a NumPy vector
def read_random_line(file_path):
    with open(file_path, 'r') as file:
        # Count the total number of lines in the file
        num_lines = sum(1 for line in file)
    
    # Generate a random line number within the range of total lines
    random_line_number = np.random.randint(0, num_lines - 1)
    
    # Read the selected random line from the file
    with open(file_path, 'r') as file:
        for line_num, line in enumerate(file):
            if line_num == random_line_number:
                # Convert the line into a NumPy vector
                numpy_vector = np.fromstring(line, dtype = float, sep=' ')
                return numpy_vector  # Return the NumPy vector
```

### Relevant log output / Error message

```shell
The output when I train on single environment:
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 80          |
|    ep_rew_mean          | -176        |
| time/                   |             |
|    fps                  | 4           |
|    iterations           | 2           |
|    time_elapsed         | 907         |
|    total_timesteps      | 4096        |
| train/                  |             |
|    approx_kl            | 0.010147858 |
|    clip_fraction        | 0.143       |
|    clip_range           | 0.2         |
|    entropy_loss         | -22.7       |
|    explained_variance   | -5.72e-06   |
|    learning_rate        | 0.0003      |
|    loss                 | 2.13e+05    |
|    n_updates            | 10          |
|    policy_gradient_loss | -0.00796    |
|    std                  | 0.997       |
|    value_loss           | 3.45e+05    |
-----------------------------------------

The output when I train on 4 environments using multiprocessing:
-----------------------------------------
| time/                   |             |
|    fps                  | 4           |
|    iterations           | 2           |
|    time_elapsed         | 907         |
|    total_timesteps      | 4096        |
-----------------------------------------
```

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by qgallouedec** [comment]
Use this instead:

```python
from stable_baselines3 import PPO
from stable_baselines3.common.env_util import make_vec_env
import gymnasium as gym


def make_env():
    return gym.make('LunarLander-v2')

env = make_vec_env(make_env, n_envs=2)

PPO('MlpPolicy', env, verbose=1).learn(1000)
```

**Comment by atharvaaalok** [comment]
@qgallouedec 
In the comment and code above
```
def make_env():
    return env
```
Here instead of the gym.make('LunarLander-v2') I just replace this with my own custom environment right?

**Comment by qgallouedec** [comment]
As written in the docs, `make_vec_env` works with any callable that returns an `gym.Env` object, so you can simply call it that way:

```python
from stable_baselines3.common.env_util import make_vec_env
import gymnasium as gym

class MyCustomEnv(gym.Env): ...

env = make_vec_env(MyCustomEnv, n_envs=2)
```

> Also here do I not need to pass any other arguments like I was passing earlier with SubprocVecEnv such as env_id:

You didn't pass `env_id` to `SubprocVecEnv` but to your own `make_env` function. Note that your `make_env` function did indeed take `env_id` but didn't use it, so you could very well have removed it. 


> I would also be grateful if you could point me to detailed examples (in documentation) of doing multiprocessing using this approach for custom environments. I wish to know how I would need to change the following function definition for this particular case:
> `def make_env(env_id: str, rank: int, seed: int = 0)`

The approach is the same as for a non-custom environment, the only thing that changes is that instead of using `gym.make`, you use `MyCustomEnv`.

If you consider that the initial question has been answered, please close the issue. If you have any further questions, please open a new one.
